File: spec/javascript/test_bytes_pad_term.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('BytesPadTerm', 'src/str_pad_term.bin', function(r) {
  assert.equal(KaitaiStream.bytesToStr(r.strPad, "UTF-8"), 'str1');
  assert.equal(KaitaiStream.bytesToStr(r.strTerm, "UTF-8"), 'str2foo');
  assert.equal(KaitaiStream.bytesToStr(r.strTermAndPad, "UTF-8"), 'str+++3bar+++');
  assert.equal(KaitaiStream.bytesToStr(r.strTermInclude, "UTF-8"), 'str4baz@');
});

File: spec/cpp_stl/test_bytes_pad_term.cpp
#include <boost/test/unit_test.hpp>

#include <bytes_pad_term.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_bytes_pad_term) {
    std::ifstream ifs("src/str_pad_term.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    bytes_pad_term_t* r = new bytes_pad_term_t(&ks);

    BOOST_CHECK_EQUAL(r->str_pad(), "str1");
    BOOST_CHECK_EQUAL(r->str_term(), "str2foo");
    BOOST_CHECK_EQUAL(r->str_term_and_pad(), "str+++3bar+++");
    BOOST_CHECK_EQUAL(r->str_term_include(), "str4baz@");

    delete r;
}
